website_sale_charge_payment_fee_delivery/models/delivery.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+from odoo import models, api
+
+
+class DeliveryCarrier(models.Model):
+    _inherit = 'delivery.carrier'
+
+    @api.multi
+    def get_price_available(self, order):
+        payment_fee_line = order.order_line.filtered('payment_fee_line')
+        if payment_fee_line:
```

Comment:
would be better to wrap the whole block inside `with self.env.do_in_draft()`

Reply:
done

Reply:
hmm I don't think is going to do what you expect the way you did it.
As soon as the ctx manager is exited you won't have `is_delivery` set and your call to `get_price_available` is going to be the same.
You should wrap the call inside it  and you don't need to set it back again ;)

Reply:
@simahawk thanks, done